# YOLACT Google Colab Notebook

# Run Inference on Video


In [2]:
file_path = "data/test.mp4"
output_path = "data/test_output.mp4"
!python eval.py --trained_model=weights/yolact_plus_resnet50_54_800000.pth --score_threshold=0.15 --top_k=15 --video={file_path}:{output_path}

Config not specified. Parsed yolact_plus_resnet50_config from the file name.

Loading model... Done.
Initializing model... Done.

Processing FPS: 0.00 | Video Playback FPS: 8.42 | Frames in Buffer: 0    %)     8.42 fps         Playback FPS: 8.42 | Frames in Buffer: 0    


# Evaluate the model pretrained on the whole 80 classes

## COCO val

Evaluate on COCO val


In [2]:
!python eval.py --trained_model=weights/yolact_plus_resnet50_54_800000.pth

Config not specified. Parsed yolact_plus_resnet50_config from the file name.

loading annotations into memory...
Done (t=0.45s)
creating index...
index created!
Loading model... Done.

Processing Images  ██████████████████████████████   4952 /   4952 (100.00%)     6.25 fps        
Saving data...
Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 34.94 | 55.41 | 53.17 | 50.49 | 47.40 | 43.20 | 37.83 | 30.48 | 20.95 |  9.30 |  1.15 |
  mask | 33.69 | 52.72 | 50.23 | 47.29 | 44.45 | 40.50 | 35.59 | 29.91 | 21.88 | 11.93 |  2.43 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+



## COCO test-dev

Evaluate on COCO test-dev


In [12]:
!python eval.py --trained_model=weights/yolact_plus_resnet50_54_800000.pth --dataset=coco2017_testdev_dataset --output_coco_json

[Errno 2] No such file or directory: '/content/yolact'
/media/max/Storage/Codice/yolact
Config not specified. Parsed yolact_plus_resnet50_config from the file name.

loading annotations into memory...
Done (t=0.38s)
creating index...
index created!
Loading model... Done.

Processing Images  ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░     64 /  20288 ( 0.32%)     3.83 fps        ^C
Stopping...


## COCO Person val

Evaluate on COCO Person val

In [1]:
!python eval.py --trained_model=weights/yolact_plus_resnet50_54_800000.pth --config=yolact_plus_resnet50_person_config --only_person

loading annotations into memory...
Done (t=0.28s)
creating index...
index created!
Loading model... Done.

Processing Images  ███░░░░░░░░░░░░░░░░░░░░░░░░░░░    302 /   2693 (11.21%)     7.81 fps        Warning: Augmentation output an example with no ground truth. Resampling...
Processing Images  █████████░░░░░░░░░░░░░░░░░░░░░    863 /   2693 (32.05%)     7.75 fps        Warning: Augmentation output an example with no ground truth. Resampling...
Processing Images  ██████████████████████████████   2693 /   2693 (100.00%)     7.11 fps        
Saving data...
Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 41.22 | 68.68 | 65.89 | 62.16 | 57.05 | 50.97 | 42.75 | 33.42 | 21.59 |  9.14 |  0.59 |
  mask | 37.45 | 64.16 | 60.82 | 56.97 | 52.01 | 46.07 | 38.70 | 28.97 | 18.23 |  7.35 |  1.20 |
-------+-------+-------+-------

# Train the model only on person class

COCO Person train has 64115 images
COCO Person val has 2693 images

In [ ]:
save_folder = "data/checkpoints/exp4"
log_folder = "data/logs/exp4"
import os
if not os.path.exists(save_folder):
  os.mkdir(save_folder)
if not os.path.exists(log_folder):
  os.mkdir(log_folder)

# Experiment 4
# Here I have modified yolact.py in order to fine-tune only the final layer.
# Prior experiments fine-tuned the whole network!
# 'max_iter': 10000
# 'lr_steps': (.35 * 10000, .75 * 10000, .88 * 10000, .93 * 10000)
!python train.py --save_folder="{save_folder}" --log_folder="{log_folder}" --config=yolact_plus_resnet50_person_config --resume=data/checkpoints/exp4/yolact_plus_resnet50_person_0_2000.pth --only_last_layer --eval_only_person --validation_iter=500 --batch_size=2 --save_interval=500 --lr=1e-5 --momentum=0.9 --decay=5e-4 --gamma=0.1

Scaling parameters by 0.25 to account for a batch size of 2.
Per-GPU batch size is less than the recommended limit for batch norm. Disabling batch norm.
loading annotations into memory...
Done (t=4.57s)
creating index...
index created!
loading annotations into memory...
Done (t=0.17s)
creating index...
index created!
Resuming training, loading data/checkpoints/exp4/yolact_plus_resnet50_person_0_2000.pth...
Begin training!

[  0]    2000 || B: 0.785 | C: 0.752 | M: 2.819 | S: 0.016 | I: 0.165 | T: 4.537 || ETA: 0:00:00 || timer: 2.996
[  0]    2010 || B: 1.083 | C: 1.157 | M: 1.914 | S: 0.038 | I: 0.131 | T: 4.323 || ETA: 5:52:52 || timer: 0.550
[  0]    2020 || B: 0.959 | C: 0.944 | M: 1.876 | S: 0.050 | I: 0.173 | T: 4.003 || ETA: 5:50:23 || timer: 0.553
[  0]    2030 || B: 1.051 | C: 1.118 | M: 2.005 | S: 0.064 | I: 0.149 | T: 4.387 || ETA: 5:46:48 || timer: 0.527
[  0]    2040 || B: 1.098 | C: 1.163 | M: 1.952 | S: 0.066 | I: 0.153 | T: 4.433 || ETA: 5:44:12 || timer: 0.523
[  0]   

[  0]    2630 || B: 1.030 | C: 1.176 | M: 1.937 | S: 0.093 | I: 0.112 | T: 4.348 || ETA: 12:44:48 || timer: 0.558
[  0]    2640 || B: 0.998 | C: 1.152 | M: 1.923 | S: 0.094 | I: 0.112 | T: 4.280 || ETA: 12:37:54 || timer: 0.533
[  0]    2650 || B: 1.044 | C: 1.177 | M: 1.963 | S: 0.097 | I: 0.115 | T: 4.396 || ETA: 12:31:27 || timer: 0.544
[  0]    2660 || B: 1.073 | C: 1.228 | M: 1.983 | S: 0.099 | I: 0.110 | T: 4.493 || ETA: 12:24:54 || timer: 0.529
[  0]    2670 || B: 1.051 | C: 1.204 | M: 1.949 | S: 0.095 | I: 0.104 | T: 4.404 || ETA: 12:18:41 || timer: 0.530
[  0]    2680 || B: 1.008 | C: 1.168 | M: 1.922 | S: 0.090 | I: 0.107 | T: 4.295 || ETA: 12:12:36 || timer: 0.553
[  0]    2690 || B: 1.026 | C: 1.201 | M: 1.958 | S: 0.089 | I: 0.108 | T: 4.382 || ETA: 12:06:40 || timer: 0.544
[  0]    2700 || B: 0.985 | C: 1.163 | M: 1.942 | S: 0.080 | I: 0.102 | T: 4.271 || ETA: 12:01:01 || timer: 0.539
[  0]    2710 || B: 0.988 | C: 1.166 | M: 1.906 | S: 0.080 | I: 0.110 | T: 4.250 || ETA:

[  0]    3290 || B: 1.222 | C: 1.291 | M: 2.184 | S: 0.062 | I: 0.122 | T: 4.881 || ETA: 14:06:12 || timer: 0.577
[  0]    3300 || B: 1.202 | C: 1.289 | M: 2.149 | S: 0.061 | I: 0.119 | T: 4.821 || ETA: 14:05:58 || timer: 0.542
[  0]    3310 || B: 1.237 | C: 1.306 | M: 2.227 | S: 0.063 | I: 0.126 | T: 4.959 || ETA: 14:05:47 || timer: 0.544
[  0]    3320 || B: 1.218 | C: 1.298 | M: 2.212 | S: 0.064 | I: 0.122 | T: 4.914 || ETA: 14:05:36 || timer: 0.570
[  0]    3330 || B: 1.224 | C: 1.260 | M: 2.165 | S: 0.060 | I: 0.123 | T: 4.831 || ETA: 14:05:20 || timer: 0.555
[  0]    3340 || B: 1.205 | C: 1.232 | M: 2.178 | S: 0.065 | I: 0.126 | T: 4.807 || ETA: 14:05:08 || timer: 0.612
[  0]    3350 || B: 1.186 | C: 1.196 | M: 2.138 | S: 0.067 | I: 0.118 | T: 4.705 || ETA: 14:04:54 || timer: 0.535
[  0]    3360 || B: 1.172 | C: 1.200 | M: 2.184 | S: 0.072 | I: 0.144 | T: 4.771 || ETA: 14:04:36 || timer: 0.531
[  0]    3370 || B: 1.072 | C: 1.128 | M: 2.058 | S: 0.071 | I: 0.140 | T: 4.469 || ETA:

[  0]    3950 || B: 1.041 | C: 1.136 | M: 1.971 | S: 0.064 | I: 0.117 | T: 4.329 || ETA: 13:24:18 || timer: 0.528
[  0]    3960 || B: 1.058 | C: 1.144 | M: 1.980 | S: 0.066 | I: 0.119 | T: 4.368 || ETA: 13:24:07 || timer: 0.533
[  0]    3970 || B: 1.087 | C: 1.190 | M: 1.966 | S: 0.069 | I: 0.126 | T: 4.437 || ETA: 13:23:51 || timer: 0.531
[  0]    3980 || B: 1.148 | C: 1.252 | M: 2.033 | S: 0.070 | T: 4.503 || ETA: 13:23:29 || timer: 0.542
[  0]    3990 || B: 1.099 | C: 1.196 | M: 1.960 | S: 0.069 | I: 0.107 | T: 4.431 || ETA: 13:23:14 || timer: 0.549
Saving state, iter: 4000

Computing validation mAP (this may take a while)...


Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 40.94 | 68.74 | 65.60 | 61.79 | 56.98 | 50.58 | 42.54 | 32.97 | 20.96 |  8.45 |  0.75 |
  mask | 37.57 | 64.48 | 61.25 | 57.27 | 52.26 | 4

[  0]    4550 || B: 1.153 | C: 1.409 | M: 2.001 | S: 0.073 | I: 0.143 | T: 4.779 || ETA: 13:03:57 || timer: 0.535
[  0]    4560 || B: 1.146 | C: 1.384 | M: 1.965 | S: 0.077 | I: 0.130 | T: 4.703 || ETA: 13:03:45 || timer: 0.533
[  0]    4570 || B: 1.090 | C: 1.324 | M: 1.860 | S: 0.070 | I: 0.139 | T: 4.483 || ETA: 13:03:26 || timer: 0.531
[  0]    4580 || B: 1.122 | C: 1.317 | M: 2.011 | S: 0.069 | I: 0.138 | T: 4.658 || ETA: 13:03:13 || timer: 0.592
[  0]    4590 || B: 1.114 | C: 1.349 | M: 2.008 | S: 0.073 | I: 0.135 | T: 4.679 || ETA: 13:03:00 || timer: 0.550
[  0]    4600 || B: 1.128 | C: 1.372 | M: 2.014 | S: 0.071 | I: 0.125 | T: 4.710 || ETA: 13:02:45 || timer: 0.533
[  0]    4610 || B: 1.101 | C: 1.333 | M: 1.961 | S: 0.071 | I: 0.119 | T: 4.584 || ETA: 13:02:26 || timer: 0.533
[  0]    4620 || B: 1.109 | C: 1.354 | M: 2.057 | S: 0.071 | I: 0.120 | T: 4.712 || ETA: 13:02:11 || timer: 0.546
[  0]    4630 || B: 1.132 | C: 1.345 | M: 2.015 | S: 0.071 | I: 0.111 | T: 4.674 || ETA:

[  0]    5210 || B: 1.163 | C: 1.194 | M: 2.010 | S: 0.067 | I: 0.161 | T: 4.595 || ETA: 12:46:31 || timer: 0.541
[  0]    5220 || B: 1.085 | C: 1.137 | M: 1.986 | S: 0.070 | I: 0.159 | T: 4.437 || ETA: 12:46:07 || timer: 0.544
[  0]    5230 || B: 1.086 | C: 1.142 | M: 1.948 | S: 0.065 | I: 0.149 | T: 4.390 || ETA: 12:45:53 || timer: 0.532
[  0]    5240 || B: 1.097 | C: 1.136 | M: 1.990 | S: 0.073 | I: 0.149 | T: 4.445 || ETA: 12:45:38 || timer: 0.539
[  0]    5250 || B: 1.097 | C: 1.111 | M: 1.996 | S: 0.072 | I: 0.151 | T: 4.427 || ETA: 12:45:23 || timer: 0.535
[  0]    5260 || B: 1.094 | C: 1.097 | M: 2.012 | S: 0.071 | I: 0.148 | T: 4.423 || ETA: 12:45:14 || timer: 0.555
[  0]    5270 || B: 1.133 | C: 1.135 | M: 2.048 | S: 0.070 | I: 0.149 | T: 4.536 || ETA: 12:45:00 || timer: 0.572
[  0]    5280 || B: 1.102 | C: 1.144 | M: 1.948 | S: 0.071 | I: 0.141 | T: 4.406 || ETA: 12:44:40 || timer: 0.534
[  0]    5290 || B: 1.077 | C: 1.150 | M: 1.922 | S: 0.072 | I: 0.107 | T: 4.328 || ETA:

[  0]    5870 || B: 1.026 | C: 1.140 | M: 1.936 | S: 0.070 | I: 0.139 | T: 4.312 || ETA: 12:23:21 || timer: 0.562
[  0]    5880 || B: 0.999 | C: 1.131 | M: 1.949 | S: 0.073 | I: 0.131 | T: 4.282 || ETA: 12:23:11 || timer: 0.545
[  0]    5890 || B: 1.032 | C: 1.126 | M: 1.981 | S: 0.072 | I: 0.144 | T: 4.355 || ETA: 12:22:56 || timer: 0.534
[  0]    5900 || B: 1.066 | C: 1.152 | M: 1.953 | S: 0.076 | I: 0.130 | T: 4.378 || ETA: 12:22:43 || timer: 0.539
[  0]    5910 || B: 1.111 | C: 1.170 | M: 2.061 | S: 0.076 | I: 0.126 | T: 4.545 || ETA: 12:22:32 || timer: 0.545
[  0]    5920 || B: 1.122 | C: 1.161 | M: 2.053 | S: 0.078 | I: 0.132 | T: 4.545 || ETA: 12:22:20 || timer: 0.563
[  0]    5930 || B: 1.101 | C: 1.140 | M: 2.080 | S: 0.082 | I: 0.125 | T: 4.528 || ETA: 12:22:07 || timer: 0.531
[  0]    5940 || B: 1.138 | C: 1.157 | M: 2.102 | S: 0.083 | I: 0.122 | T: 4.603 || ETA: 12:21:55 || timer: 0.554
[  0]    5950 || B: 1.081 | C: 1.149 | M: 2.035 | S: 0.088 | I: 0.131 | T: 4.484 || ETA:

[  0]    6500 || B: 1.056 | C: 1.121 | M: 2.058 | S: 0.073 | I: 0.111 | T: 4.420 || ETA: 12:30:22 || timer: 405.211
[  0]    6510 || B: 1.048 | C: 1.151 | M: 2.052 | S: 0.074 | I: 0.122 | T: 4.447 || ETA: 12:30:07 || timer: 0.531
[  0]    6520 || B: 1.051 | C: 1.174 | M: 2.107 | S: 0.072 | I: 0.130 | T: 4.533 || ETA: 12:29:53 || timer: 0.550
[  0]    6530 || B: 1.044 | C: 1.181 | M: 2.131 | S: 0.074 | I: 0.190 | T: 4.620 || ETA: 12:29:40 || timer: 0.532
[  0]    6540 || B: 1.069 | C: 1.197 | M: 2.181 | S: 0.073 | I: 0.223 | T: 4.743 || ETA: 12:29:25 || timer: 0.522
[  0]    6550 || B: 1.086 | C: 1.184 | M: 2.395 | S: 0.073 | I: 0.221 | T: 4.959 || ETA: 12:29:13 || timer: 0.538
[  0]    6560 || B: 1.015 | C: 1.123 | M: 2.306 | S: 0.075 | I: 0.224 | T: 4.743 || ETA: 12:28:59 || timer: 0.556
[  0]    6570 || B: 1.020 | C: 1.091 | M: 2.310 | S: 0.066 | I: 0.221 | T: 4.709 || ETA: 12:28:43 || timer: 0.553
[  0]    6580 || B: 1.061 | C: 1.129 | M: 2.317 | S: 0.067 | I: 0.210 | T: 4.784 || ET

[  0]    7160 || B: 1.036 | C: 1.133 | M: 1.866 | S: 0.076 | I: 0.121 | T: 4.232 || ETA: 12:17:07 || timer: 0.552
[  0]    7170 || B: 1.073 | C: 1.201 | M: 1.870 | S: 0.073 | I: 0.123 | T: 4.339 || ETA: 12:16:55 || timer: 0.564
[  0]    7180 || B: 1.085 | C: 1.194 | M: 1.920 | S: 0.070 | I: 0.122 | T: 4.391 || ETA: 12:16:40 || timer: 0.532
[  0]    7190 || B: 1.020 | C: 1.171 | M: 1.871 | S: 0.068 | I: 0.102 | T: 4.231 || ETA: 12:16:27 || timer: 0.550
[  0]    7200 || B: 0.997 | C: 1.194 | M: 1.835 | S: 0.065 | I: 0.094 | T: 4.185 || ETA: 12:16:14 || timer: 0.564
[  0]    7210 || B: 0.951 | C: 1.163 | M: 1.829 | S: 0.065 | I: 0.089 | T: 4.097 || ETA: 12:16:03 || timer: 0.550
[  0]    7220 || B: 1.028 | C: 1.198 | M: 1.874 | S: 0.061 | I: 0.105 | T: 4.267 || ETA: 12:15:47 || timer: 0.544
[  0]    7230 || B: 1.002 | C: 1.180 | M: 1.917 | S: 0.060 | I: 0.101 | T: 4.260 || ETA: 12:15:32 || timer: 0.526
[  0]    7240 || B: 1.003 | C: 1.195 | M: 1.884 | S: 0.056 | I: 0.096 | T: 4.234 || ETA:

[  0]    7820 || B: 0.956 | C: 1.110 | M: 2.005 | S: 0.074 | I: 0.144 | T: 4.289 || ETA: 12:09:13 || timer: 0.541
[  0]    7830 || B: 0.947 | C: 1.099 | M: 1.966 | S: 0.075 | I: 0.129 | T: 4.217 || ETA: 12:09:01 || timer: 0.529
[  0]    7840 || B: 0.950 | C: 1.103 | M: 2.013 | S: 0.077 | I: 0.129 | T: 4.272 || ETA: 12:08:49 || timer: 0.547
[  0]    7850 || B: 0.975 | C: 1.130 | M: 2.057 | S: 0.079 | I: 0.132 | T: 4.373 || ETA: 12:08:35 || timer: 0.550
[  0]    7860 || B: 1.032 | C: 1.168 | M: 2.112 | S: 0.075 | I: 0.140 | T: 4.527 || ETA: 12:08:19 || timer: 0.540
[  0]    7870 || B: 1.060 | C: 1.204 | M: 2.088 | S: 0.076 | I: 0.141 | T: 4.570 || ETA: 12:08:04 || timer: 0.542
[  0]    7880 || B: 1.100 | C: 1.265 | M: 2.070 | S: 0.090 | I: 0.157 | T: 4.682 || ETA: 12:07:49 || timer: 0.536
[  0]    7890 || B: 1.113 | C: 1.274 | M: 2.049 | S: 0.079 | I: 0.132 | T: 4.647 || ETA: 12:07:37 || timer: 0.552
[  0]    7900 || B: 1.084 | C: 1.271 | M: 2.049 | S: 0.077 | I: 0.141 | T: 4.621 || ETA:

[  0]    8480 || B: 1.082 | C: 1.258 | M: 1.980 | S: 0.074 | I: 0.157 | T: 4.552 || ETA: 11:55:55 || timer: 0.566
[  0]    8490 || B: 1.097 | C: 1.291 | M: 2.049 | S: 0.077 | I: 0.154 | T: 4.668 || ETA: 11:55:41 || timer: 0.534
Saving state, iter: 8500

Computing validation mAP (this may take a while)...


Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 41.06 | 68.68 | 65.42 | 61.69 | 57.28 | 50.67 | 42.61 | 33.31 | 21.30 |  8.81 |  0.80 |
  mask | 37.47 | 64.48 | 61.22 | 57.38 | 52.15 | 45.82 | 38.33 | 28.85 | 17.99 |  7.29 |  1.17 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

[  0]    8500 || B: 1.140 | C: 1.291 | M: 2.137 | S: 0.072 | I: 0.159 | T: 4.800 || ETA: 11:53:55 || timer: 415.029
[  0]    8510 || B: 1.151 | C: 1.298 | M: 2.136 | S: 0.067 | I: 0.156 

[  0]    9070 || B: 1.243 | C: 1.253 | M: 2.161 | S: 0.060 | I: 0.119 | T: 4.834 || ETA: 11:42:20 || timer: 0.538
[  0]    9080 || B: 1.249 | C: 1.255 | M: 2.181 | S: 0.058 | I: 0.115 | T: 4.858 || ETA: 11:42:03 || timer: 0.540
[  0]    9090 || B: 1.198 | C: 1.230 | M: 2.123 | S: 0.057 | I: 0.109 | T: 4.718 || ETA: 11:41:50 || timer: 0.566
[  0]    9100 || B: 1.159 | C: 1.205 | M: 2.085 | S: 0.057 | I: 0.097 | T: 4.603 || ETA: 11:41:30 || timer: 0.539
[  0]    9110 || B: 1.140 | C: 1.175 | M: 2.144 | S: 0.061 | I: 0.114 | T: 4.634 || ETA: 11:41:17 || timer: 0.562
[  0]    9120 || B: 1.157 | C: 1.174 | M: 2.203 | S: 0.061 | I: 0.118 | T: 4.713 || ETA: 11:41:07 || timer: 0.563
[  0]    9130 || B: 1.038 | C: 1.087 | M: 1.957 | S: 0.064 | I: 0.118 | T: 4.263 || ETA: 11:40:55 || timer: 0.551
[  0]    9140 || B: 1.055 | C: 1.084 | M: 1.976 | S: 0.064 | I: 0.126 | T: 4.306 || ETA: 11:40:42 || timer: 0.562
[  0]    9150 || B: 1.032 | C: 1.047 | M: 1.974 | S: 0.064 | I: 0.141 | T: 4.259 || ETA:

[  0]    9730 || B: 1.034 | C: 1.183 | M: 1.986 | S: 0.065 | I: 0.146 | T: 4.414 || ETA: 11:36:07 || timer: 0.527
[  0]    9740 || B: 1.072 | C: 1.205 | M: 1.972 | S: 0.065 | I: 0.140 | T: 4.453 || ETA: 11:35:49 || timer: 0.551
[  0]    9750 || B: 1.080 | C: 1.204 | M: 2.007 | S: 0.062 | I: 0.140 | T: 4.493 || ETA: 11:35:35 || timer: 0.528
[  0]    9760 || B: 1.063 | C: 1.203 | M: 2.013 | S: 0.066 | I: 0.147 | T: 4.492 || ETA: 11:35:20 || timer: 0.536
[  0]    9770 || B: 1.062 | C: 1.195 | M: 1.996 | S: 0.065 | I: 0.134 | T: 4.452 || ETA: 11:35:04 || timer: 0.540
[  0]    9780 || B: 1.049 | C: 1.148 | M: 1.964 | S: 0.061 | I: 0.126 | T: 4.348 || ETA: 11:34:50 || timer: 0.552
[  0]    9790 || B: 1.097 | C: 1.161 | M: 2.023 | S: 0.053 | I: 0.116 | T: 4.451 || ETA: 11:34:36 || timer: 0.536
[  0]    9800 || B: 1.043 | C: 1.131 | M: 1.991 | S: 0.054 | I: 0.104 | T: 4.324 || ETA: 11:34:18 || timer: 0.553
[  0]    9810 || B: 1.046 | C: 1.142 | M: 2.040 | S: 0.055 | I: 0.100 | T: 4.383 || ETA:

[  0]   10390 || B: 0.860 | C: 0.999 | M: 1.761 | S: 0.063 | I: 0.128 | T: 3.810 || ETA: 11:24:42 || timer: 0.563
[  0]   10400 || B: 0.937 | C: 1.045 | M: 1.876 | S: 0.064 | I: 0.113 | T: 4.034 || ETA: 11:24:20 || timer: 0.531
[  0]   10410 || B: 0.963 | C: 1.056 | M: 1.911 | S: 0.065 | I: 0.113 | T: 4.108 || ETA: 11:24:02 || timer: 0.557
[  0]   10420 || B: 0.938 | C: 1.080 | M: 1.875 | S: 0.067 | I: 0.106 | T: 4.066 || ETA: 11:23:46 || timer: 0.550
[  0]   10430 || B: 0.931 | C: 1.057 | M: 1.842 | S: 0.066 | I: 0.106 | T: 4.003 || ETA: 11:23:32 || timer: 0.564
[  0]   10440 || B: 0.958 | C: 1.085 | M: 1.848 | S: 0.061 | I: 0.099 | T: 4.051 || ETA: 11:23:19 || timer: 0.527
[  0]   10450 || B: 0.985 | C: 1.125 | M: 1.884 | S: 0.058 | I: 0.099 | T: 4.150 || ETA: 11:23:04 || timer: 0.533
[  0]   10460 || B: 0.975 | C: 1.122 | M: 1.861 | S: 0.059 | I: 0.096 | T: 4.113 || ETA: 11:22:51 || timer: 0.527
[  0]   10470 || B: 0.988 | C: 1.169 | M: 1.868 | S: 0.058 | I: 0.097 | T: 4.180 || ETA:

[  0]   11000 || B: 1.124 | C: 1.267 | M: 1.986 | S: 0.081 | I: 0.110 | T: 4.569 || ETA: 11:07:19 || timer: 416.390
[  0]   11010 || B: 1.091 | C: 1.213 | M: 1.919 | S: 0.080 | I: 0.110 | T: 4.413 || ETA: 11:07:04 || timer: 0.531
[  0]   11020 || B: 1.024 | C: 1.146 | M: 1.849 | S: 0.080 | I: 0.111 | T: 4.210 || ETA: 11:06:49 || timer: 0.527
[  0]   11030 || B: 0.984 | C: 1.130 | M: 1.828 | S: 0.068 | I: 0.113 | T: 4.122 || ETA: 11:06:28 || timer: 0.564
[  0]   11040 || B: 0.938 | C: 1.106 | M: 1.768 | S: 0.069 | I: 0.106 | T: 3.988 || ETA: 11:06:14 || timer: 0.531
[  0]   11050 || B: 0.930 | C: 1.094 | M: 1.731 | S: 0.069 | I: 0.099 | T: 3.923 || ETA: 11:05:59 || timer: 0.540
[  0]   11060 || B: 0.928 | C: 1.110 | M: 1.734 | S: 0.075 | I: 0.098 | T: 3.944 || ETA: 11:05:41 || timer: 0.529
[  0]   11070 || B: 0.892 | C: 1.075 | M: 1.698 | S: 0.073 | I: 0.106 | T: 3.845 || ETA: 11:05:22 || timer: 0.531
[  0]   11080 || B: 0.914 | C: 1.178 | M: 1.726 | S: 0.070 | I: 0.108 | T: 3.995 || ET

[  0]   11660 || B: 1.134 | C: 1.161 | M: 1.996 | S: 0.077 | I: 0.103 | T: 4.472 || ETA: 10:44:07 || timer: 0.529
[  0]   11670 || B: 1.128 | C: 1.160 | M: 2.016 | S: 0.075 | I: 0.109 | T: 4.488 || ETA: 10:43:54 || timer: 0.532
[  0]   11680 || B: 1.152 | C: 1.185 | M: 2.071 | S: 0.075 | I: 0.134 | T: 4.617 || ETA: 10:43:40 || timer: 0.530
[  0]   11690 || B: 1.154 | C: 1.217 | M: 2.106 | S: 0.076 | I: 0.153 | T: 4.706 || ETA: 10:43:27 || timer: 0.548
[  0]   11700 || B: 1.136 | C: 1.192 | M: 2.125 | S: 0.074 | I: 0.148 | T: 4.674 || ETA: 10:43:15 || timer: 0.535
[  0]   11710 || B: 1.063 | C: 1.149 | M: 2.024 | S: 0.083 | I: 0.151 | T: 4.471 || ETA: 10:43:01 || timer: 0.541
[  0]   11720 || B: 1.040 | C: 1.106 | M: 1.957 | S: 0.072 | I: 0.145 | T: 4.321 || ETA: 10:42:47 || timer: 0.535
[  0]   11730 || B: 1.015 | C: 1.104 | M: 1.938 | S: 0.071 | I: 0.155 | T: 4.284 || ETA: 10:42:34 || timer: 0.541
[  0]   11740 || B: 0.971 | C: 1.087 | M: 1.849 | S: 0.072 | I: 0.148 | T: 4.127 || ETA:

[  0]   12320 || B: 1.130 | C: 1.256 | M: 2.011 | S: 0.079 | I: 0.114 | T: 4.589 || ETA: 10:27:58 || timer: 0.534
[  0]   12330 || B: 1.103 | C: 1.258 | M: 1.990 | S: 0.078 | I: 0.116 | T: 4.546 || ETA: 10:27:46 || timer: 0.554
[  0]   12340 || B: 1.146 | C: 1.292 | M: 2.018 | S: 0.075 | I: 0.120 | T: 4.650 || ETA: 10:27:33 || timer: 0.537
[  0]   12350 || B: 1.130 | C: 1.252 | M: 1.983 | S: 0.071 | I: 0.117 | T: 4.553 || ETA: 10:27:20 || timer: 0.532
[  0]   12360 || B: 1.061 | C: 1.200 | M: 1.890 | S: 0.074 | I: 0.113 | T: 4.338 || ETA: 10:27:06 || timer: 0.533
[  0]   12370 || B: 1.054 | C: 1.161 | M: 1.904 | S: 0.068 | I: 0.110 | T: 4.297 || ETA: 10:26:53 || timer: 0.543
[  0]   12380 || B: 1.111 | C: 1.194 | M: 1.953 | S: 0.073 | I: 0.113 | T: 4.444 || ETA: 10:26:37 || timer: 0.537
[  0]   12390 || B: 1.132 | C: 1.260 | M: 1.997 | S: 0.068 | I: 0.111 | T: 4.568 || ETA: 10:26:25 || timer: 0.547
[  0]   12400 || B: 1.063 | C: 1.189 | M: 1.915 | S: 0.067 | I: 0.110 | T: 4.344 || ETA:

[  0]   12980 || B: 0.900 | C: 1.177 | M: 1.809 | S: 0.072 | I: 0.116 | T: 4.074 || ETA: 10:27:57 || timer: 0.538
[  0]   12990 || B: 0.908 | C: 1.164 | M: 1.835 | S: 0.070 | I: 0.128 | T: 4.104 || ETA: 10:27:43 || timer: 0.537
Saving state, iter: 13000

Computing validation mAP (this may take a while)...


Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 41.65 | 69.09 | 66.28 | 62.54 | 57.57 | 51.39 | 43.50 | 34.15 | 21.73 |  8.77 |  1.45 |
  mask | 37.85 | 64.19 | 61.22 | 57.49 | 52.69 | 46.73 | 39.34 | 29.68 | 18.51 |  7.47 |  1.18 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

[  0]   13000 || B: 0.901 | C: 1.136 | M: 1.754 | S: 0.063 | I: 0.103 | T: 3.956 || ETA: 10:41:37 || timer: 445.213
[  0]   13010 || B: 0.898 | C: 1.130 | M: 1.793 | S: 0.060 | I: 0.104

[  0]   13570 || B: 0.988 | C: 1.093 | M: 1.954 | S: 0.071 | I: 0.118 | T: 4.223 || ETA: 10:29:38 || timer: 0.539
[  0]   13580 || B: 0.969 | C: 1.108 | M: 1.875 | S: 0.068 | I: 0.118 | T: 4.137 || ETA: 10:29:22 || timer: 0.533
[  0]   13590 || B: 0.966 | C: 1.129 | M: 1.914 | S: 0.067 | I: 0.124 | T: 4.200 || ETA: 10:29:10 || timer: 0.544
[  0]   13600 || B: 0.959 | C: 1.148 | M: 1.924 | S: 0.069 | I: 0.128 | T: 4.227 || ETA: 10:28:56 || timer: 0.542
[  0]   13610 || B: 0.992 | C: 1.127 | M: 1.968 | S: 0.070 | I: 0.124 | T: 4.282 || ETA: 10:28:40 || timer: 0.527
[  0]   13620 || B: 0.992 | C: 1.147 | M: 1.984 | S: 0.072 | I: 0.126 | T: 4.321 || ETA: 10:28:22 || timer: 0.536
[  0]   13630 || B: 1.036 | C: 1.181 | M: 2.032 | S: 0.075 | I: 0.133 | T: 4.459 || ETA: 10:28:06 || timer: 0.550
[  0]   13640 || B: 1.043 | C: 1.194 | M: 2.018 | S: 0.075 | I: 0.134 | T: 4.463 || ETA: 10:27:53 || timer: 0.547
[  0]   13650 || B: 0.963 | C: 1.165 | M: 1.946 | S: 0.076 | I: 0.120 | T: 4.270 || ETA:

# Evaluate the model trained only on the person class

Evaluate on COCO Person val

In [9]:
!python eval.py --trained_model=weights/yolact_plus_resnet50_person_0_1400.pth --config=yolact_plus_resnet50_person_config --only_person

Config not specified. Parsed yolact_plus_resnet50_person_config from the file name.

loading annotations into memory...
Done (t=0.28s)
creating index...
index created!
Loading model... Done.

Processing Images  ███░░░░░░░░░░░░░░░░░░░░░░░░░░░    302 /   2693 (11.21%)     6.98 fps        Warning: Augmentation output an example with no ground truth. Resampling...
Processing Images  █████████░░░░░░░░░░░░░░░░░░░░░    863 /   2693 (32.05%)     7.17 fps        Warning: Augmentation output an example with no ground truth. Resampling...
Processing Images  ██████████████████████████████   2693 /   2693 (100.00%)     8.03 fps        
Saving data...
Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 38.91 | 66.20 | 63.26 | 59.25 | 54.42 | 48.26 | 40.28 | 31.00 | 19.30 |  6.81 |  0.35 |
  mask | 35.79 | 62.37 | 58.91 | 55.23 | 50